src/overlays/actors/ovl_En_Hit_Tag/z_en_hit_tag.c

```diff
@@ -29,21 +28,66 @@ const ActorInit En_Hit_Tag_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BE21F0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 16, 32, 0, { 0, 0, 0 } },
 };
 
-#endif
+void EnHitTag_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderCylinder* pCylinder;
+    EnHitTag* this = (EnHitTag*)thisx;
 
-extern ColliderCylinderInit D_80BE21F0;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = func_80BE20E8;
+    pCylinder = &this->cylinder;
+    Collider_InitAndSetCylinder(globalCtx, &this->cylinder, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->cylinder);
+    if (Flags_GetSwitch(globalCtx, (s32)(this->actor.params & 0xFE00) >> 9)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hit_Tag/EnHitTag_Init.s")
+void EnHitTag_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHitTag* this = (EnHitTag*)thisx;
+    Collider_DestroyCylinder(globalCtx, &this->cylinder);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hit_Tag/EnHitTag_Destroy.s")
+void func_80BE20E8(EnHitTag* this, GlobalContext* globalCtx) {
```

Comment:
Documentation is not usually required, but for such a simple actor here we could probably change the name of this function now, something like `EnHitTag_ActionFunc` since there is only one or `EnHitTag_WaitForHit` or `EnHitTag_Idle`.

To rename you can use the tool in `tools/rename_sym.sh` to change the name of a function,  where the first argument is the symbol you want to replace, and the second is what you want to replace with, example:
`rename_sym.sh func_80BE20E8 EnHitTag_Idle` should replace it for all instances in the repo

Reply:
Thanks, I took your suggestion of EnHitTag_WaitForHit